packages/fuels-programs/src/contract.rs

```diff
@@ -49,12 +100,91 @@ pub trait SettableContract {
     fn log_decoder(&self) -> LogDecoder;
 }
 
-/// A compiled representation of a contract.
+/// A compiled representation of a contract
 #[derive(Debug, Clone, Default)]
 pub struct CompiledContract {
-    pub raw: Vec<u8>,
-    pub salt: Salt,
-    pub storage_slots: Vec<StorageSlot>,
+    binary: Vec<u8>,
+    salt: Salt,
+    storage_slots: Vec<StorageSlot>,
+}
+
+/// Configuration for contract storage
+#[derive(Debug, Clone, Default)]
+pub struct StorageConfiguration {
+    storage_path: String,
+    manual_storage: Vec<StorageSlot>,
+}
+
+impl StorageConfiguration {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn new_from(storage_path: String, manual_storage: Vec<StorageSlot>) -> Self {
+        Self {
+            storage_path,
+            manual_storage,
+        }
+    }
+
+    pub fn set_storage_path(mut self, storage_path: String) -> Self {
+        self.storage_path = storage_path;
+        self
+    }
+
+    pub fn set_manual_storage(mut self, manual_storage: Vec<StorageSlot>) -> Self {
+        self.manual_storage = manual_storage;
+        self
+    }
+}
+
+/// Configuration for contract deployment
+#[derive(Debug, Clone, Default)]
+pub struct DeployConfiguration {
+    tx_parameters: TxParameters,
+    storage: StorageConfiguration,
+    configurables: Configurables,
+    salt: Salt,
+}
+
+impl DeployConfiguration {
+    pub fn new() -> Self {
```

Comment:
IMO it would make more sense for `new()` to be the current `new_from()` and let default be `default()`.
Using `new()` to init a default seems obscure.

Reply:
I missed the above comment, but I think this still stands. Just because `default()` is not used in the wild doesn't mean we shouldn't encourage it.

Reply:
It is very common to have `new()` without arguments making a default struct. So much so, that clippy has a warning for it https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default

Reply:
I agree with @MujkicA here.

I.e. `new` takes all the struct args and `default` provides sensible defaults.

Reply:
+1

This is relatively minor, IMO. Still, a nice touch 😄